onnxruntime/core/providers/cpu/ml/svmclassifier.cc

```diff
@@ -99,24 +117,34 @@ Status SVMClassifier<T>::Compute(OpKernelContext* ctx) const {
   {
     int64_t current_weight_0 = n * stride;
     int64_t maxclass = -1;
-    double maxweight = 0.f;
     std::vector<float> decisions;
     std::vector<float> scores;
     std::vector<float> kernels;
     std::vector<int64_t> votes;
+    float sum;
+
+    if (vector_count_ == 0 && mode_ == SVM_TYPE::SVM_LINEAR) {
+      // This was in the original code but it does not appear in libsvm or scikit-learn.
+      for (int64_t j = 0; j < class_count_; j++) {  //for each class
+        float val = kernel_dot(x_data, current_weight_0, coefficients_, feature_count_ * j,
+                               feature_count_, get_kernel_type());
+        val += rho_[0];
+        scores.push_back(val);
+      }
+    } else {
+      if (vector_count_ == 0)
+        return Status(common::ONNXRUNTIME, common::FAIL, "No support vectors.");
+      int evals = 0;
 
-    if (mode_ == SVM_TYPE::SVM_SVC) {
       for (int64_t j = 0; j < vector_count_; j++) {
-        float val = kernel_dot(x_data, current_weight_0, support_vectors_, feature_count_ * j, feature_count_, get_kernel_type());
+        float val = kernel_dot(x_data, current_weight_0, support_vectors_, feature_count_ * j,
+                               feature_count_, get_kernel_type());
         kernels.push_back(val);
       }
-      for (int64_t j = 0; j < class_count_; j++) {
-        votes.push_back(0);
-      }
-      int evals = 0;
-      for (int64_t i = 0; i < class_count_; i++) {        //for each class
-        for (int64_t j = i + 1; j < class_count_; j++) {  //for each class
-          float sum = 0;
+      votes.resize(class_count_, 0);
+      for (int64_t i = 0; i < class_count_; i++) {        // for each class
```

Comment:
May we do `std::vector<int64_t> votes(class_count_)` when declaring it?

Reply:
It is done with a resize but it is not always resized.